src/Share/Codebase/CodeCache.hs

```diff
@@ -161,12 +194,55 @@ getTypeDeclsByRefIdsOf codeCache@(CodeCache {codeCacheCodebaseEnv}) trav s = do
                     Nothing -> (mempty, Nothing)
                 Right decl -> (mempty, Just decl)
 
-      cacheDecls codeCache cacheable
-      pure $ hydrated'
+      cacheDecls codeCache (Map.fromList cacheable)
+      pure hydrated'
+
+expectTypeDeclsByRefIdsOf ::
+  (QueryM m) =>
+  CodeCache scope ->
+  Traversal s t TypeReferenceId (V1.Decl Symbol Ann) ->
+  s ->
+  m t
+expectTypeDeclsByRefIdsOf codeCache@(CodeCache {codeCacheCodebaseEnv}) trav s = do
```

Comment:
What do you think of just calling directly into `getTypeDeclsByRefIdsOf` then unwrapping the Maybe's into an error rather than copying the implementation? I think it'd make it easier to update these methods in the future :)

Reply:
That seems fine; the inner database calls here are:

- `Codebase.loadV1TypeDeclarationsByRefIdsOf : Id -> Maybe Decl`
- `Codebase.expectTypeDeclarationsByRefIdsOf : Id -> Decl`

so if we don't want to have a variant at the code cache level for an underlying call to `Codebase.expectTypeDeclarationsByRefIdsOf`, then we do have do duplicate the error it throws when a decl is missing. That is only a tiny amount of duplication, though. Still in favor of doing that instead of what's currently in the PR?